src/screens/map-view-screen.tsx

```diff
@@ -163,16 +186,55 @@ export const MapViewTest = () => {
                     <Slider.FilledTrack />
                 </Slider.Track>
                 <Slider.Thumb />
-            </Slider>
-            <Text>{distanceLabel}Km</Text>
-            <Text>{location && `Location on Load, ${location[0]}, ${location[1]}`}</Text>
+            </Slider> */}
+
+            {/* <Text>{distanceLabel}Km</Text> */}
             <Text>
-                {location &&
-                    regionChange &&
-                    `Changing Location, ${regionChange.latitude.toFixed(
+                {initialLocation && `Your Location, ${initialLocation[0]}, ${initialLocation[1]}`}
+            </Text>
+            <Text>
+                {regionChange &&
+                    `Map Location, ${regionChange.latitude.toFixed(
                         6,
                     )}, ${regionChange.longitude.toFixed(6)}`}
             </Text>
+
+            <GooglePlacesAutocomplete
+                styles={{
+                    container: {
+                        flex: 0,
+                        width: '100%',
+                    },
+                    textInput: {
+                        fontSize: 18,
+                    },
+                }}
+                textInputProps={{
+                    returnKeyType: 'string',
+                }}
+                minLength={2}
+                enablePoweredByContainer={false}
+                placeholder="Search Bar"
+                nearbyPlacesAPI="GooglePlacesSearch"
+                debounce={400}
+                onPress={(data, details = null) => {
+                    if (details) {
+                        setCurrentLocation({
+                            latitude: details?.geometry.location.lat,
+                            longitude: details?.geometry.location.lng,
+                            latitudeDelta: 0.0922,
+                            longitudeDelta: 0.0421,
+                        });
+                    }
+                }}
+                fetchDetails={true}
+                onFail={(error) => console.error(error)}
+                query={{
+                    key: GOOGLE_API_KEY,
+                    language: 'en',
+                }}
+            />
+
```

Comment:
It might be better to make this GooglePlacesAutocomplete component inside of its own search bar or something. This will help you to not have to constantly import this Google_Api_Key throughout all of the code. Additionally, you will probably have similar props wherever you use this in the app, so it would be better to implement it once, then add it to whatever component in the future.